### PR TITLE
React redux

### PR DESCRIPTION
**Pull request by robinbenitezmora**
We are giving much more functionality to the app by adding books.

### CODE REVIEW

File: src/components/book/Book.js [hunk 1]
```diff
@@ -1,21 +1,49 @@
...
+import { removeBook } from '../../redux/books/books';
+
+const Book = (props) => {
+  const dispatch = useDispatch();
+  const { book } = props;
+  const { id, title, author } = book;
+
+  const handleRemoveBook = (book) => {
+    dispatch(removeBook(book.id));
+  };
```
Comment (ShahierNashaat) [inline src/components/book/Book.js]: - [ ] Kindly notice that the remove action is not working you need to remove the `book` parameter from `handleRemoveBook` to make remove work.

```javascript
const handleRemoveBook = () => {
    dispatch(removeBook(book.id));
};
```

File: src/components/book/Book.js [hunk 2]
```diff
@@ -1,21 +1,49 @@
...
+  },
+};
+
+Book.propTypes = {
+  book: PropTypes.shape({
+    id: PropTypes.number,
+    title: PropTypes.string,
+    author: PropTypes.string,
+  }),
+};
```
Comment (ShahierNashaat) [inline src/components/book/Book.js]: - [ ] You have made a great job working on a unique id for the book but you should notice that the id is a string and you made its type to be a number so you should change that.

```javascript
Book.defaultProps = {
  book: {
    id: '',
    title: '',
    author: '',
  },
};

Book.propTypes = {
  book: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    author: PropTypes.string,
  }),
};
```

File: src/redux/books/books.js [hunk 3]
```diff
@@ -4,21 +4,39 @@ const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
...
+    author: 'J. R. R. Tolkien',
+    category: 'Fantasy',
+  },
+  {
+    id: 6,
+    title: 'A Brief History of Time',
+    author: 'Stephen Hawking',
+    category: 'Sci-Fi',
   },
 ];
```
Comment (ShahierNashaat) [inline src/redux/books/books.js]: - [ ] Also here you have to make the ids string as they should be like the books that you will add.

```javascript
const firstState = [
  {
    id: '1',
    title: 'The Hunger Games',
    author: 'Suzanne Collins',
    category: 'Action',
  },
  {
    id: '2',
    title: 'Dune',
    author: 'Frank Herbert',
    category: 'Sci-Fi',
  },
  {
    id: '3',
    title: 'Capital in the Twenty-First Century',
    author: 'Suzanne Collins',
    category: 'Economy',
  },
  {
    id: '4',
    title: 'The Lord of the Rings',
    author: 'J. R. R. Tolkien',
    category: 'Fantasy',
  },
  {
    id: '5',
    title: 'The Hobbit',
    author: 'J. R. R. Tolkien',
    category: 'Fantasy',
  },
  {
    id: '6',
    title: 'A Brief History of Time',
    author: 'Stephen Hawking',
    category: 'Sci-Fi',
  },
];
```

File: src/pages/Categories.js [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
 
-function Categories() {
   return (
-    <button type="button">Check Status</button>
+    <button type="button" className="btn btn-primary" onClick={handleFilter}>Check Status</button>
   );
-}
+};
 
 export default Categories;
```
Comment (ShahierNashaat) [inline src/pages/Categories.js]: - [ ] Kindly use the `checkStatus` method that dispatch the check status action also you should get the categories state to display it.

```javascript
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { checkStatus } from '../redux/categories/categories';

const Categories = () => {
  const category = useSelector((state) => state.categories);
  const dispatch = useDispatch();
  const handleFilter = (e) => {
    dispatch(checkStatus(e.target.value));
  };

  return (
    <>
      <button type="button" className="btn btn-primary" onClick={handleFilter}>Check Status</button>
      <p>{category}</p>
    </>
  );
};

export default Categories;
```

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @robinbenitezmora, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — APPROVED** [review]
Hi @robinbenitezmora 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**